Update file connector example page (#1222)
wso2/docs-mi: en/docs/reference/connectors/file-connector/file-connector-example.md

    
    <img src="{{base_path}}/assets/img/integrate/connectors/filecon1.png" title="create project" width="500" alt="create project"/>

### Create the API

1.  Create an API named `TestAPI` with the `/fileConnector` context. This API will accept employee information.

    <img src="{{base_path}}/assets/img/integrate/connectors/filecon2.png" title="create API" width="500" alt="create API"/>

2. Change the `Url Style` to `URI_TEMPLATE` and update the value of `Uri-Template` to `/create`.

> 3. In the default resource of the API, enable POST requests. 

Reviewer: ```suggestion
3. In the default resource of the API, enable `POST` requests. 
```

    <img src="{{base_path}}/assets/img/integrate/connectors/filecon3.png" title="select post method" width="500" alt="select post method"/>

> ### Add a Log Mediator

Reviewer: ```suggestion
### Add a Log mediator
```

1.  Add the Log mediator to the design canvas and configure a custom log that indicates when the API receives a message.  

> ### Add and Configure the Datamapper Mediator to convert XML to CSV

Reviewer: ```suggestion
### Add and configure the Data Mapper mediator to convert XML to CSV
```

> 1.  Add the DataMapper mediator and configure it to transform the incoming XML message to a CSV message. 

Reviewer: ```suggestion
1.  Add the Data Mapper mediator and configure it to transform the incoming XML message to a CSV message. 
```

> 2.  Double-click the Datamapper mediator and add a new transform configuration called 'xmlToCsvConfig'. Provide an existing Registry resource or create a new one to provide as the 

Reviewer: ```suggestion
2.  Double-click the Data Mapper mediator and add a new transform configuration called `xmlToCsvConfig`. Provide an existing Registry resource or create a new one to provide as the 
```

    </test>
    ```

> 4.  Load the input file into the Datamapper config view.

Reviewer: ```suggestion
4.  Load the input file into the Data Mapper config view.
```

    <img src="{{base_path}}/assets/img/integrate/connectors/filecon5.png" title="load input" width="800" alt="load input"/>

5.  Save the following content as a CSV file. This will be the data output file.

    ```
    Name,Age,Company
    Hasitha,34,wso2
    Johan,32,IBM
    ```

> 6. Load the output CSV file into the datamapper config view.

Reviewer: ```suggestion
6. Load the output CSV file into the Data Mapper config view.
```

    <img src="{{base_path}}/assets/img/integrate/connectors/filecon6.png" title="load output csv" width="800" alt="load output csv"/>

7.  Configure the mapping as shown below. Each element in the input should be mapped to the respective element in the output. 

    <img src="{{base_path}}/assets/img/integrate/connectors/filecon7.png" title="data mapping" width="800" alt="data mapping"/>

> 8.  Specify the input as XML and output as CSV in the datamapper as shown below. 

Reviewer: ```suggestion
8.  Specify the input as XML and output as CSV in the Data Mapper as shown below. 
```

    <img src="{{base_path}}/assets/img/integrate/connectors/filecon8.png" title="datamapper input output config" width="800" alt="datamapper input output config"/>

> ### Add an Enrich Mediator

Reviewer: ```suggestion
### Add an Enrich mediator
```

> 1.  Add the Enrich mediator and configure it to save the output generated by the datamapper to a property named `CONTENT`. 

Reviewer: ```suggestion
1.  Add the Enrich mediator and configure it to save the output generated by the Data Mapper to a property named `CONTENT`. 
```

    <img src="{{base_path}}/assets/img/integrate/connectors/filecon9.png" title="enrich - save payload" width="800" alt="enrich - save payload"/>

> ### Use File Connector Operations

Reviewer: ```suggestion
### Use File connector operations
```

        <img src="{{base_path}}/assets/img/integrate/connectors/filecon12.png" title="filter mediator" width="800" alt="filter mediator"/>

> 3. Inside the “else” block of the Filter mediator, add the File Connector's <b>write</b> operation and configure it to write the static content of CSV file headers: “Name,Age,Company”. 

Reviewer: ```suggestion
3. Inside the `else` block of the Filter mediator, add the File connector's <b>write</b> operation and configure it to write the static content of CSV file headers: `Name,Age,Company`. 
```

    2. Configure the file path as `/dataCollection/employees/employees.csv`.
    3. The <b>Write Mode</b> needs to be `Append`.
    4. Set the <b>Position<b> to `2`. This will always append the message to line number 2 as we need the newest message at the top.
    5. Set <b>Append New Line</b> to `true`. 

    <img src="{{base_path}}/assets/img/integrate/connectors/filecon15.png" title="append to file" width="800" alt="append to file"/>

6. Add the File connector’s <b>read</b> operation and configure it to read the same CSV file. 

    1. Add the file connection created in the earlier steps as the file connection for this operation.
    2. Configure the file path as `/dataCollection/employees/employees.csv`.
    3. Set the <b>Read Mode</b> to `Starting From Line` and the <b>Start Line Number</b> to `2`.
    4. Set the <b>Content Type<b> to `text/plain`. 

    !!! Note
        The file reading will start from line number 2. The content will be read as a text message. 

    <img src="{{base_path}}/assets/img/integrate/connectors/filecon16.png" title="read csv file" width="800" alt="read csv file"/>

> ### Use the DataMapper Mediator to Convert CSV to XML

Reviewer: ```suggestion
### Use the Data Mapper mediator to convert CSV to XML
```

> 1. Add the Datamapper mediator again and configure it to convert the CSV message (after reading) back to XML.

Reviewer: ```suggestion
1. Add the Data Mapper mediator again and configure it to convert the CSV message (after reading) back to XML.
```

>     1.  Double-click the data mapper and add a new configuration called 'csvToXmlConfig'.

Reviewer: ```suggestion
    1.  Double-click the data mapper and add a new configuration called `csvToXmlConfig`.
```

        <img src="{{base_path}}/assets/img/integrate/connectors/filecon17.png" title="output datamapper config" width="800" alt="output datamapper config"/>

    2.  This time, the mapping should be from CSV to XML.

        <img src="{{base_path}}/assets/img/integrate/connectors/filecon18.png" title="output datamapper dialog" width="800" alt="output datamapper dialog"/>

2. Finally, use the <b>Respond</b> mediator to send the transformed message to the API caller. 

> ### Configure a Fault Sequence

Reviewer: ```suggestion
### Configure a fault sequence
```

## Testing

> 1. Create a file called data.xml with the following payload. 

Reviewer: ```suggestion
1. Create a file called `data.xml` with the following payload. 
```
